### PR TITLE
Switching to minilzo to fix 64bit builds for load/save

### PR DESCRIPTION
**Pull request by capsterx**
Switched to minilzo.
Made sure to LZO1X_1_MEM_COMPRESS instead of 64 * 1024.  It worked with the hardcoded value but valgrind complained it was overwriting data.
changed unsigned int to lzo_uint (was required)
changed unsigned char * to lzo_bytep (was not required, but figured it was better to do)

### CODE REVIEW

File: redalert/CMakeLists.txt [hunk 1]
```diff
@@ -270,4 +269,4 @@ if(BUILD_VANILLARA)
     if(MSVC)
         set_target_properties(VanillaRA PROPERTIES LINK_FLAGS "/subsystem:windows /ENTRY:WinMainCRTStartup")
     endif()
-endif()
```
Comment (hifi) [inline redalert/CMakeLists.txt:273]: Avoid EOL changes at the end of a file. If you use Vim (I do), add `set nofixendofline` to your `.vimrc`. It isn't a big issue but it creates unnecessary churn to PRs when editors fight over it.

**Review by hifi — CHANGES_REQUESTED** [review]
It would probably be the right move to replace the aging LZO implementation with a more modern one to fix the underlying issue. However, LZO compression is only used for saves so I would prefer we rip it off completely and use uncompressed saves or change it to standard zlib and link against it over adding more code that we need to maintain.

If uncompressed saves are small enough we get a major savings in code maintenance to just remove compression completely. Saves will be reimplemented in a more portable way later on anyway so keeping better LZO implementation around would only be a short term solution.

**Comment by hifi** [comment]
Since we already have LCWPipe and LCWStraw at hand and we need to carry them forever as the game depends on the compression I opened #237 that outright removes LZO and uses LCW instead.

Sorry I'm shooting down all of your PRs but they have touching things that I've either been working on already or I've decided on another approach completely like this one.

**Comment by capsterx** [comment]
I was originally trying to keep save files compatible, but now that I think about it, your right.  The way it dumps the raw binary wouldnt be compatible so there is no reason to have to use lzo.  Probably better to serialize the save file in a portable way instead of just dumping the binary objects as they do.

**Comment by hifi** [comment]
We have a long term plan to serialize a portable versioned file using the pipe/straw system that takes endianness into account. That should be a permanent fix and enable moving saves across platforms but it will take a while to implement and for both games.

**Pull request closed without merging**